File: upl-purchase-sdk-php/src/Model/ResponseWithAuthorization.php
<?php

/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 */

declare(strict_types=1);

namespace Unzer\PayLater\Model;

use Unzer\PayLater\Communication\Response;

class ResponseWithAuthorization implements Response
{
    /**
     * @var string
     */
    private $authorization;

    /**
     * @var Response
     */
    private $response;

    /**
     */
    public function __construct(string $authorization, Response $response)
    {
        $this->authorization = $authorization;
        $this->response = $response;
    }

    /**
     * @return string
     */
    public function getAuthorization(): string
    {
        return $this->authorization;
    }

    /**
     * @return Response
     */
    public function getResponse(): Response
    {
        return $this->response;
    }
}

File: upl-purchase-sdk-java/src/main/java/com/unzer/paylater/model/ResponseWithAuthorization.java
package com.unzer.paylater.model;

public class ResponseWithAuthorization<O> {
    private final String authorization;
    private final O response;

    public ResponseWithAuthorization(String authorization, O response) {
        this.authorization = authorization;
        this.response = response;
    }

    public String getAuthorization() {
        return authorization;
    }

    public O getResponse() {
        return response;
    }
}
